contracts/Subscription.sol

```diff
@@ -1,205 +1,356 @@
+                       INTERNAL CONSTANT FUNCTIONS
+    //////////////////////////////////////////////////////////////*/
+
+    /// @inheritdoc ERC5643
+    function _isRenewable() internal view override returns (bool) {
         return _renewable;
     }
 
-    /**
-     * @dev Gets the price to renew a subscription for `duration` seconds for
-     * a given tokenId. This should be overridden in implementing contracts.
-     */
-    // solhint-disable-next-line
-    function _getRenewalPrice(uint256 tokenId, uint64 duration) internal view virtual returns (uint256) {
-        return 0.1 ether;
+    /// @inheritdoc ERC5643
+    function _getRenewalPrice(uint64 duration) internal view override returns (uint256) {
+        return duration * nativeCurrencyPrice;
     }
 
-    /**
-     * @dev Internal function to set the minimum renewal duration.
-     */
-    function _setMinimumRenewalDuration(uint64 duration) internal virtual {
-        _minimumRenewalDuration = duration;
+    /*//////////////////////////////////////////////////////////////
+                     INTERNAL NON-CONSTANT FUNCTIONS
+    //////////////////////////////////////////////////////////////*/
+
+    /// @dev Internal function to handle ERC20 token payment for subscription and extend subscription.
+    /// @param tokenId The unique identifier of the subscription token.
+    /// @param duration The duration (in seconds) to extend the subscription for.
+    /// @param erc20 The address of the ERC20 token to use for payment.
+    function _handleERC20PaymentSubscriptionUpdate(uint256 tokenId, uint64 duration, address erc20) internal {
+        _validateDurationBetweenMinAndMax(duration);
+
+        // ERC-20 Specific pre-flight checks
+        _handleERC20Payment(duration, erc20);
+
+        // extend subscription
+        _updateSubscriptionExpiration(tokenId, duration);
     }
 
-    /**
-     * @dev Internal function to set the maximum renewal duration.
-     */
-    function _setMaximumRenewalDuration(uint64 duration) internal virtual {
-        _maximumRenewalDuration = duration;
+    /// @dev Internal function to handle ERC20 token payment with pre-flight checks.
+    /// @param duration The duration (in seconds) to extend the subscription for.
+    /// @param erc20 The address of the ERC20 token to use for payment.
+    function _handleERC20Payment(uint64 duration, address erc20) internal {
+        address msgSender = msg.sender;
+        address addressThis = address(this);
+
+        uint256 tokensQtyToTransfer = chargeAmountForERC20(erc20) * duration;
+        IERC20 payableToken = IERC20(erc20);
+
+        if (payableToken.balanceOf(msgSender) < tokensQtyToTransfer) revert ERC20InsufficientBalance();
+        if (payableToken.allowance(msgSender, addressThis) < tokensQtyToTransfer) {
+            revert ERC20InsufficientAllowance();
+        }
+
+        ERC20TransferHelper.safeTransferFrom(IERC20(erc20), msgSender, addressThis, tokensQtyToTransfer);
     }
 }
```

Comment:
what function does an owner of a cre8or call to extend their subscription?

Looks like all the methods are `minterOnly`

Can we remove all unused methods for 8/8 mint day and focus on MVP functionality? It's cool you're thinking through all the bells & whistles but let's focus on keeping this simple.